### DISCUSSION TITLE
法语支持

### DISCUSSION THREAD

**Discussion by BUENDIA1994** [hiroi-sora/Umi-OCR#360] [Ideas]
能否在ocr识别中增加法语库，我可以测试审阅

**Comment by hiroi-sora** [comment]
PaddleOCR 项目组 [提供的模型库列表](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_ch/models_list.md#23-%E5%A4%9A%E8%AF%AD%E8%A8%80%E8%AF%86%E5%88%AB%E6%A8%A1%E5%9E%8B%E6%9B%B4%E5%A4%9A%E8%AF%AD%E8%A8%80%E6%8C%81%E7%BB%AD%E6%9B%B4%E6%96%B0%E4%B8%AD) 中，暂时未包含法语模型。因此，本项目的Paddle和Rapid引擎插件均不支持法语。

训练一个新模型需要训练数据等成本，我个人也暂无法提供新语言模型的训练。

不过，Umi-OCR支持另一个引擎插件：TesseractOCR。该 [插件](https://github.com/qwedc001/tesseractOCR_umi_plugin) 目前正在开发完善中，将支持手动导入法语模型库。

（TesseractOCR插件目前的 v1.0.0 版本有一些文字位置错误的小问题，修复中）

**Reply by qwedc001** [reply]
检查了一下，PPv3 确实没有法语，但是PPv2有
他们的80种小语种的说法应该是已经实装了，但是是旧版不是新版
[下载链接](https://aistudio.baidu.com/modelsdetail/16/download)

**Comment by qwedc001** [comment]
你好，这里是 TesseractOCR插件 的作者。TesseractOCR原项目提供了法语相关的模型文件，您可以下载下来使用。
针对于用户的使用教程在 [这里](https://github.com/qwedc001/tesseractOCR_umi_plugin?tab=readme-ov-file#%E5%AF%B9%E4%BA%8E%E7%94%A8%E6%88%B7) 可以找到。
由于TesseractOCR语言库较多，出于精简插件体积的考虑，我仅内置了中英日语言库，法语需要额外下载。
可以前往在插件库中提到的 [Tesseract_Fast](https://github.com/tesseract-ocr/tessdata_fast) 或者 [Tesseract_Best](https://github.com/tesseract-ocr/tessdata_best) 中下载 `fra.traineddata` 文件，并按照文档添加进文件夹即可被程序识别到。

**Reply by BUENDIA1994** [reply]
经过测试，法语好用，目前没有发现问题！

**Reply by Oceannagirl** [reply]
好像所有的à,ê都无法识别

**Reply by qwedc001** [reply]
> 好像所有的à,ê都无法识别

请参考issues 常见问题FAQ

**Reply by BUENDIA1994** [reply]
https://github.com/hiroi-sora/Umi-OCR/issues/447#issuecomment-2025245747 识别率低没有办法，目前我用chatgpt对识别后的文本进行校正，这样的识别错误是可以改过来的。
如果是文本生成的pdf是可以准确识别的，参考这个测试文本
[法文特殊字符.pdf](https://github.com/hiroi-sora/Umi-OCR/files/15050521/default.pdf)
